apps/mull-api/src/app/channel/channel.service.ts

```diff
@@ -1,26 +1,59 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
-import { Channel } from '../entities';
-import { CreateChannelInput } from './inputs/channel.input';
+import { Channel, DirectMessageChannel, EventChannel } from '../entities';
+import { CreateDmChannelInput, CreateEventChannelInput } from './inputs/channel.input';
 @Injectable()
 export class ChannelService {
   constructor(
     @InjectRepository(Channel)
-    private channelRepository: Repository<Channel>
+    private channelRepository: Repository<Channel>,
+    @InjectRepository(EventChannel)
+    private eventChannelRepository: Repository<EventChannel>,
+    @InjectRepository(DirectMessageChannel)
+    private dmChannelRepository: Repository<DirectMessageChannel>
   ) {}
 
   getChannel(channelId: number): Promise<Channel> {
     return this.channelRepository.findOne(channelId);
   }
 
-  async createChannel(input: CreateChannelInput, eventId?: number): Promise<boolean> {
-    await this.channelRepository.save({ ...input, event: { id: eventId } });
+  getEventChannel(channelId: number): Promise<EventChannel> {
+    return this.eventChannelRepository.findOne(channelId);
+  }
+
+  getDmChannel(channelId: number): Promise<DirectMessageChannel> {
+    return this.dmChannelRepository.findOne(channelId, { relations: ['participants'] });
+  }
+
+  async createEventChannel(input: CreateEventChannelInput, eventId: number): Promise<boolean> {
+    await this.eventChannelRepository.save({ ...input, event: { id: eventId } });
     return true;
   }
 
+  async createDmChannel(input: CreateDmChannelInput): Promise<DirectMessageChannel> {
+    return this.dmChannelRepository.save(input);
+  }
+
   async deleteChannel(channelId: number): Promise<boolean> {
     await this.channelRepository.delete(channelId);
     return true;
   }
+
+  async findUsersInDmChannel(userId1: number, userId2: number): Promise<DirectMessageChannel> {
+    const dmChannelsList = await this.dmChannelRepository
+      .createQueryBuilder()
+      .leftJoinAndSelect('DirectMessageChannel.participants', 'participants')
+      .where('participants.id = :userId1 OR participants.id = :userId2', { userId1, userId2 })
+      .getMany();
+
+    // TODO: redo this filtering as TypeORM syntax
+    return dmChannelsList.find(({ participants }) => {
+      let count = 0;
+      for (const participant of participants) {
+        if (participant.id === userId1 || participant.id === userId2) count++;
+      }
+      return count === 2;
```

Comment:
This isn't a blocker just maybe a suggestion if we have time to refactor this. 

Can we have a column that concatenates the users ids to find the channel with a separator in the middle?  
Say we have user with id 1 and user with id 2, then we add to the database the string `1+2`. say we have a user with id 111 and user with id 11 then we would have the string `11+111`. we would always have the smallest number first .

Maybe it's a bad idea I'm not sure. Just wanted to hear your opinions

Reply:
After talking, we decided not to do this.